lib/pbench/server/cache_manager.py

```diff
@@ -1025,39 +1037,13 @@ def subprocess_run(
             raise exception(ctx, str(exc)) from exc
         else:
             if process.returncode != 0:
-
-                # tar failures can include a message for each file in the
-                # archive, and can be quite large. Break stderr into lines, and
-                # gather those lines only up to our configured size limit.
-                #
-                # The exception we raise and ultimately log will start by
-                # identifying the tarball, followed by the command and the
-                # return code. The appended stderr may provide additional
-                # useful context, but in general beyond the first line it'll
-                # be fairly meaningless.
-                size = 0
-                message = []
-                lines = process.stderr.split("\n")
-                prefix = ""
-                for line in lines:
-                    # Skip empty lines
-                    if not line:
-                        continue
-                    size += len(line)
-                    if size > MAX_ERROR:
-                        prefix = "[TRUNC]"
-                        break
-                    message.append(line)
-
-                # Prefix with [TRUNC] if we're not including the entire message
-                # and truncate the first line if it was too long by itself.
-                if message:
-                    msg = prefix + ("\n".join(message))
-                else:
-                    msg = "[TRUNC]" + lines[0][:MAX_ERROR]
+                msg = process.stderr
+                if len(msg) > MAX_ERROR - len(TRUNC_PREFIX):
+                    msg = (TRUNC_PREFIX + msg)[:MAX_ERROR]
```

Comment:
I don't think the length of the truncation marker needs to enter into the conditional:  if the message is not larger than `MAX_ERROR`, then we don't need to truncate it (even if it's within `len(TRUNC_PREFIX)` of the limit).  The expression at line 1042 will properly handle the situation if it needs truncating, and we can leave it alone if it doesn't.

Reply:
True. Two editing passes cross in the night while the captain sleeps, and "stuff happens" ... 😦